builtin/logical/pki/path_tidy.go

```diff
@@ -2114,6 +2111,24 @@ func (b *backend) tidyStatusIncCMPV2NonceDeletedCount() {
 	b.tidyStatus.cmpv2NonceDeletedCount++
 }
 
+// updateLastAutoTidyTime should be used to update b.lastAutoTidy as the required locks
+// are acquired and the auto tidy time is persisted to storage to work across restarts
+func (b *backend) updateLastAutoTidyTime(sc *storageContext, lastRunTime time.Time) error {
+	b.tidyStatusLock.Lock()
+	defer b.tidyStatusLock.Unlock()
+
+	b.lastAutoTidy = lastRunTime
+	return sc.writeAutoTidyLastRun(lastRunTime)
+}
+
+// getLastAutoTidyTime should be used to read from b.lastAutoTidy as the required locks
+// are acquired prior to reading
+func (b *backend) getLastAutoTidyTime() time.Time {
+	b.tidyStatusLock.RLock()
+	defer b.tidyStatusLock.RUnlock()
+	return b.lastAutoTidy
```

Comment:
I think this might show some really odd behavior around startup - in particular implying that tidy has been run much more recently than it has been, and with a default window of 5 minutes, that's... a long time for this to be wrong.  Could we check that lastAutoTidy != startup time here, and return an error if they are equal?

Reply:
It shouldn't be wrong for long as within the mount's initialize function we reset it to either `Now()`, if it was never persisted, or from the loaded value from disk. See https://github.com/hashicorp/vault/pull/28488/files#diff-55509815a5c3c783c81fbd7e9a4cb4cda87e48750d68c740871a633c1de090a0R467

The use case that we never persisted before, we can't really do much, and technically it matches what we previously did as it was set to `Now()` every startup.

Reply:
Initialize happens before tidy, oops :) - 
At the very least, this isn't a regression, so resolving.